src/xrpld/rpc/handlers/LedgerEntry.cpp

```diff
@@ -821,7 +821,7 @@ parsePermissionedDomains(Json::Value const& pd, Json::Value& jvResult)
         return std::nullopt;
     }
 
-    if (!pd.isMember(jss::account) || !pd[jss::account].isString())
```

Comment:
why is the string check removed?

Reply:
You are right, I misinterpreted how method `asString()` works.
I created a separate `if` to check account filed type and add a new test when account is an object.